src/prefect/engine/task_runner.py

```diff
@@ -152,12 +148,12 @@ def initialize_run(  # type: ignore
                 "task_loop_count": state.cached_inputs.pop(  # type: ignore
                     "_loop_count"
                 )  # type: ignore
-                .to_result()
+                .populate_result(self.task.result)  # todo: here
```

Comment:
I'm a little confused how/why this case is different from a similar case below (https://github.com/PrefectHQ/prefect/pull/2307/files#diff-7e1afdbbda2b3755cc338f22cc84d9dcR972). On the discussion PR there was a suggestion that the serializers handle this instead somehow (https://github.com/PrefectHQ/prefect/pull/2278#discussion_r404311132). Not feeling confident about what exactly this is doing so would like to get some more info on that.